instrumentation/shoryuken/lib/opentelemetry/instrumentation/shoryuken/middlewares/server/tracer_middleware.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module OpenTelemetry
+  module Instrumentation
+    module Shoryuken
+      module Middlewares
+        module Server
+          # TracerMiddleware propagates context and instruments Shoryuken requests
+          # by way of its middleware system
+          class TracerMiddleware
+            def call(worker_instance, queue, sqs_msg, _body) # rubocop:disable Metrics/MethodLength
+              attributes = {
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_SYSTEM => 'shoryuken',
+                'messaging.shoryuken.job_class' => worker_instance.class.name,
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_MESSAGE_ID => sqs_msg.message_id,
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_DESTINATION => queue,
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_DESTINATION_KIND => 'queue',
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_OPERATION => 'process'
+              }
+              if sqs_msg.respond_to?(:attributes)
+                sqs_msg.attributes.each do |k, v|
+                  attributes["messaging.attributes.#{k}"] = v
```

Comment:
Are these arbitrary values in the message payload? I.e. is this recording the contents of the message as span attributes?

Reply:
Yes, I was following the approach that the Datadog instrumentation took. See: https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/tracing/contrib/shoryuken/tracer.rb#L39

Reply:
I am surprised that dd-trace does this.

AFAIK None of the instrumentations capture values in message payloads or request bodies.

I'm worried about having high cardinality values or exposing values in the message that contain sensitive data and PII that users are unable to suppress. 

Although this will be asymmetric from dd-trace I would like to request that you omit this functionality.

Reply:
No worries. Can understand the concern, will remove.

Reply:
Sorry, I looked at the data I can see where we have this code running in production and it looks like they aren't _arbitrary_ values. Looks like it's standard message SQS metadata, see screenshot below:

<img width="805" alt="Screen Shot 2023-07-04 at 10 08 34 am" src="https://github.com/open-telemetry/opentelemetry-ruby-contrib/assets/950337/0e37c308-8465-4eeb-bc5b-211bf5f9b9c8">

This would make sense of the DD approach. If you're concerned about controlling the cardinality I could change the implementation to support the known list of attributes?

Reply:
Thanks for patiently waiting on a reply. 

I think that this should be limited to the well known upstream semconv attributes. 
https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/messaging/#messaging-attributes

It there are attributes that are specific to AWS SQS then I think they should either be suppressed or If you have time please take a look at the semconv repo for any PRs/discussions around AWS SQS specific metadata  as well as review what other language SIGs are doing for these attributes. 

I suspect that the latter would result in attribute not covered by the messaging spec to be namespaced under `messaging.aws.sqs` or something similar.

Reply:
Fair enough. Had a quick scan but couldn't see any examples of doing anything with these sqs specific attributes so have removed.